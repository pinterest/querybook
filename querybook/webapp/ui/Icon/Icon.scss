.Icon {
    display: inline-flex;

    &.fill {
        svg {
            fill: currentColor;
        }
    }

    svg.feather-loader {
        animation: loading 3s linear forwards infinite;
    }
}

@keyframes loading {
    100% {
        transform: rotate(360deg);
    }
}
