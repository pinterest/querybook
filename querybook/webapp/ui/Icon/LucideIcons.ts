/**
 * All Lucide Icons used in Querybook
 *
 * Please make sure the list is Alphabetically sorted
 * When adding, add to both import and export
 *
 */
import {
    Activity,
    AlertCircle,
    AlertOctagon,
    AlignCenterHorizontal,
    AlignCenterVertical,
    AlignLeft,
    ArrowDown,
    ArrowLeft,
    ArrowRight,
    ArrowUp,
    Bold,
    Book,
    BookOpen,
    Box,
    Briefcase,
    Check,
    CheckCircle,
    ChevronDown,
    ChevronLeft,
    ChevronRight,
    ChevronsDown,
    ChevronsUp,
    ChevronUp,
    Circle,
    Clipboard,
    Clock,
    Code,
    Command,
    Copy,
    Crown,
    Database,
    Delete,
    Download,
    Edit,
    Edit2,
    Edit3,
    Expand,
    ExternalLink,
    Eye,
    EyeOff,
    File,
    FileOutput,
    FileText,
    Filter,
    FormInput,
    HelpCircle,
    Home,
    Info,
    Italic,
    Key,
    Languages,
    Layout,
    Link,
    List,
    ListOrdered,
    Lock,
    LogOut,
    Maximize,
    Maximize2,
    Menu,
    Minimize2,
    Minus,
    MoreVertical,
    MoveVertical,
    Network,
    PieChart,
    Play,
    PlayCircle,
    PlugZap,
    Plus,
    Quote,
    RefreshCw,
    Repeat,
    Save,
    Scissors,
    Search,
    Server,
    Settings,
    Share,
    Sidebar,
    Slash,
    Sliders,
    Star,
    Strikethrough,
    Terminal,
    ThumbsDown,
    ThumbsUp,
    Trash,
    Trash2,
    Type,
    Underline,
    Unlock,
    Upload,
    UserMinus,
    UserPlus,
    Users,
    Volume2,
    Watch,
    X,
    XCircle,
    Zap,
} from 'lucide-react';

const AllLucideIcons = {
    Activity,
    AlertCircle,
    AlertOctagon,
    AlignCenterVertical,
    AlignCenterHorizontal,
    AlignLeft,
    ArrowDown,
    ArrowLeft,
    ArrowRight,
    ArrowUp,
    Bold,
    Book,
    BookOpen,
    Box,
    Briefcase,
    Check,
    CheckCircle,
    ChevronDown,
    ChevronLeft,
    ChevronRight,
    ChevronUp,
    ChevronsDown,
    ChevronsUp,
    Circle,
    Clipboard,
    Clock,
    Code,
    Copy,
    Command,
    Crown,
    Database,
    Delete,
    Download,
    Edit,
    Edit2,
    Edit3,
    Eye,
    EyeOff,
    Expand,
    ExternalLink,
    File,
    FileOutput,
    FileText,
    Filter,
    FormInput,
    HelpCircle,
    Home,
    Info,
    Italic,
    Key,
    Languages,
    Layout,
    Link,
    List,
    ListOrdered,
    Loading: Circle, // Custom loading icon
    Lock,
    LogOut,
    Maximize,
    Maximize2,
    Menu,
    Minimize2,
    Minus,
    MoreVertical,
    Network,
    MoveVertical,
    PieChart,
    Play,
    PlayCircle,
    PlugZap,
    Plus,
    Quote,
    RefreshCw,
    Repeat,
    Save,
    Scissors,
    Search,
    Server,
    Settings,
    Sidebar,
    Share,
    Slash,
    Sliders,
    Star,
    Strikethrough,
    Terminal,
    ThumbsDown,
    ThumbsUp,
    Trash,
    Trash2,
    Type,
    Underline,
    Unlock,
    Upload,
    UserMinus,
    UserPlus,
    Users,
    Volume2,
    Watch,
    X,
    XCircle,
    Zap,
};

export type AllLucideIconNames = keyof typeof AllLucideIcons;
export default AllLucideIcons;
