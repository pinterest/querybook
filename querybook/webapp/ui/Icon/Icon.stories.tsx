import React from 'react';
// eslint-disable-next-line import/no-extraneous-dependencies
import centered from '@storybook/addon-centered/react';

import { Icon, IIconProps } from './Icon';

export const _Icon = (args: IIconProps) => <Icon {...args} />;

_Icon.args = {
    size: 16,
    name: 'feather',
    fill: false,
};

_Icon.argTypes = {
    size: {
        control: {
            type: 'range',
            min: 10,
            max: 50,
        },
    },
};

export default {
    title: 'Stateless/Icon',
    decorators: [centered],
};
