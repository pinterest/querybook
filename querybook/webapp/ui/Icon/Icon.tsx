import React from 'react';
import feather, { FeatherAttributes } from 'feather-icons';

import './Icon.scss';

// Wrapper for feather icon
export interface IIconProps {
    className?: string;
    size?: string | number;
    name: string;
    options?: FeatherAttributes;
    fill?: boolean;
}

export const Icon: React.FunctionComponent<IIconProps> = React.memo(
    ({ name, className = '', size, options = {}, fill = false }) => {
        if (!(name in feather.icons)) {
            return null;
        }
        if (size != null) {
            options.width = size;
            options.height = size;
        }

        const rawSvg = feather.icons[name].toSvg(options);

        return (
            <span
                className={`${className} Icon ${fill ? 'fill' : ''}`}
                dangerouslySetInnerHTML={{ __html: rawSvg }}
            />
        );
    }
);
