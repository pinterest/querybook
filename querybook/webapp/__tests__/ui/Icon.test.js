import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import renderer from 'react-test-renderer';

import { Icon } from '../../ui/Icon/Icon';

it('renders without crashing', () => {
    mount(<Icon name="Check" />);
});

describe('matches enzyme snapshots', () => {
    it('matches snapshot', () => {
        const wrapper = mount(<Icon name="Check" />);
        const serialized = toJson(wrapper);
        expect(serialized).toMatchSnapshot();
    });
});

describe('matches test renderer snapshot', () => {
    it('serializes the styles', () => {
        const output = renderer.create(<Icon name="Check" />);
        expect(output).toMatchSnapshot();
    });
});
