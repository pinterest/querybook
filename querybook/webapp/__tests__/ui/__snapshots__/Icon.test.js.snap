// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches enzyme snapshots matches snapshot 1`] = `
<Memo()
  name="Check"
>
  <span
    className="Icon"
  >
    <Check>
      <svg
        className="lucide lucide-check"
        fill="none"
        height={24}
        stroke="currentColor"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth={2}
        viewBox="0 0 24 24"
        width={24}
        xmlns="http://www.w3.org/2000/svg"
      >
        <polyline
          key="10jjfj"
          points="20 6 9 17 4 12"
        />
      </svg>
    </Check>
  </span>
</Memo()>
`;

exports[`matches test renderer snapshot serializes the styles 1`] = `
<span
  className="Icon"
>
  <svg
    className="lucide lucide-check"
    fill="none"
    height={24}
    stroke="currentColor"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth={2}
    viewBox="0 0 24 24"
    width={24}
    xmlns="http://www.w3.org/2000/svg"
  >
    <polyline
      points="20 6 9 17 4 12"
    />
  </svg>
</span>
`;
